### PR TITLE
Modernize js to es6

### PR DESCRIPTION
**Pull request by pilafmon**
1. Set `strict` to `"implied"` (updated `this` to `global`)
2. `var` to `const` or `let`
3. functions to fat arrows (`=>`) and had to move definitions before usage
4. Bumped testing timeout from 1000 to 1500 to allow completion
5. Updated **.jshintrc** and added **npm** `pretest` script to do linting

This PR changes a lot of lines of code, but the changes are intended to be just syntactic updates.  There should be no functional differences at all.  The code should now look more modern but still behave exactly the same.  I attempted to follow the existing styles in the project (such as indentation and semicolon usage) as much as possible, but feel free to reformat any style I didn't get right.

### CODE REVIEW

File: .gitignore
Comment (stevemao) [inline .gitignore]: Can you ignore yarn.lock here and put `package-lock=false` in .npmrc?

File: test.js
Comment (stevemao) [inline test.js]: Why do we need this comment here?

Comment (pilafmon) [reply test.js]: It's not needed... it's just a very minimal header at the top of the file to give the developer a tiny bit of context when they are working on multiple projects at the same time.  If you don't like it, it totally can be removed no problem.

**Review by stevemao — APPROVED** [review]
Thanks! LGTM

File: index.js
Comment (grncdr) [inline index.js]: this should be `return output`

File: index.js
Comment (grncdr) [inline index.js]: same as above

**Review by grncdr — CHANGES_REQUESTED** [review]

**Comment by grncdr** [comment]
I don't want to discourage contribution, but I don't like this PR for the following reasons:

1. Unrelated changes combined into a single commit (the changes to config files are a separate concern from the code changes).
2. It accidentally breaks the API.
3. No explanation for _why_ to make the change is given.
4. It changes and re-orders nearly every line of code, which combined with points 1 and point 3 makes reviewing it feel like a chore.

I'll defer to the other maintainers if they want to fix the second point and merge anyways, but IMO this PR should be rejected.

File: .gitignore
Comment (pilafmon) [reply .gitignore]: Done.

File: index.js
Comment (pilafmon) [reply index.js]: Done.

File: index.js
Comment (pilafmon) [reply index.js]: Done.

**Pull request closed without merging**
